RELEASE.md

```diff
@@ -0,0 +1,14 @@
+# Release Process
+
+1. Merge PR.
+2. Open a new PR to update CHANGELOG ([example](https://github.com/fastly/terraform-provider-fastly/pull/348)).
```

Comment:
Would be good to seem some additional notes here about our changelog conventions. I.e. we follow semver and only include relevant/significant changes in the log etc.

Reply:
I'd take a look at the CLI or Go Fastly docs for inspiration: https://github.com/fastly/go-fastly/blob/master/RELEASE.md#how-to-cut-a-new-release-for-the-go-fastly-http-client